### ISSUE TITLE
mozilla like and firefox sometimes freeze

### ISSUE THREAD

**Issue by madovsky** [i-rinat/freshplayerplugin#306]
My system is Fedora 23 64 bits on an USB key.
everything is working well but randomly sometimes everything is frozen and the system just logout after 3 mn

**Comment by i-rinat** [comment]
Looks like driver issue.

Due to its nature, freshplayerplugin makes offscreen compositing, and I've seen intel GPU hang in some cases. You can try to disable XRender by adding `enable_xrender = 0` to `~/.config/freshwrapper.conf`. Or disable 3d completely by adding `enable_3d = 0` to the same file.

In no case should whole system hang due to an application activity. If however that happens, it's drivers that should be fixed, not the application.

**Comment by madovsky** [comment]
ok understood. I'm going to test your suggestion thanks

**Comment by madovsky** [comment]
ok the issue didn't come from the GPU driver but apparently from the other issue I opened
about the RTMP and RTMFP real time streaming
if I select the v4l2loopback and start to broadcast, it freezes the mouse and can't do anything.
after flash showed a stream exception

**Comment by madovsky** [comment]
Is it possible that it can come from the sound card?
I used enable_xrender to 0 but no success and noticed that when it freezes I hear a little "sound click", the same sound  when the sound card is intialized

**Comment by i-rinat** [comment]
> Is it possible that it can come from the sound card?

Could be. Although clicking sound could be unrelated at all.

I've seen a notebook which freezed when one started playing Youtube. But after all the culprit was not a graphics or a sound card, but a Wi-Fi driver. So it's not always obvious where the bug is. Fortunately, the driver was fixed eventually.

There are multiple ways of debugging that. One is to monitor Linux kernel log via serial console. Or with netconsole, which is almost the same, but works though Ethernet. Another way is so called "crash kernel": a block in memory is reserved for another copy of kernel. When main kernel panics, it jumps into the crash kernel. Since there will be no hard reset, main kernel memory content is preserved and could be dumped to further debugging.

I used latter technique to find out `dmesg` content. It didn't help me to fix it, but at least I found out where the bug was.
